replica/database.cc

```diff
@@ -1411,6 +1411,10 @@ bool database::can_apply_per_partition_rate_limit(const schema& s, db::operation
     return replica::can_apply_per_partition_rate_limit(s, _dbcfg, op_type);
 }
 
+bool database::is_internal_query() const {
+    return classify_request(_dbcfg) != request_class::user;
+}
+
```

Comment:
Will this be called on the coordinator or replica?

I'm a little worried that if it is called on the replica, different replicas will reach different decisions and we'll get confused.

Reply:
Coordinator. This is called by `storage_proxy::get_tombstone_limit()`, which is called by callers of coordinator side `storage_proxy::query()`.